workflow/scripts/add_extra_components.py

```diff
@@ -119,6 +130,103 @@ def attach_storageunits(n, costs, elec_opts, investment_year):
         )
 
 
+def attach_phs_storageunits(n: pypsa.Network, elec_opts):
```

Comment:
I think the function looks great! Just to clarify though, we are attaching the PHS in `extra_components` rather than in `add_electricity` just due to the amount of data that comes with the PHS datasets?

Reply:
Not that I disagree with the placement! I just know we attach existing batteries in `add_electricity`, and other storage options here. So I guess we add in all PHS here since existing PHS facilities is much smaller than resource potential?